packages/bot/src/scratch/index.js

```diff
@@ -81,6 +81,21 @@ export const runBot = (limitations = {}) => {
             block.setDisabled(true);
         }
     });
+
+    const blocks_in_workspace = Blockly.derivWorkspace.blockDB_;
```

Comment:
- You can use `Blockly.derivWorkspace.getAllBlocks()` to get all the blocks, try to minimise the use of internal functions/properties (those marked with an underscore) as much as possible.